sdks/python/apache_beam/io/mongodbio.py

```diff
@@ -732,13 +750,18 @@ def process(self, element, *args, **kwargs):
       # distributed across too many processes. See more on the ObjectId format
       # https://docs.mongodb.com/manual/reference/bson-types/#objectid.
       element["_id"] = objectid.ObjectId()
-
     yield element
 
 
 class _WriteMongoFn(DoFn):
   def __init__(
-      self, uri=None, db=None, coll=None, batch_size=100, extra_params=None):
+      self,
+      uri=None,
+      db=None,
+      coll=None,
+      batch_size=100,
+      extra_params=None,
+      writeFn=None):
```

Comment:
We can define the typehint of write_func argument as Optional[str, typing.Callable]. If it is None or "ReplaceOne" then set write_func= _ReplaceOneWriteFunc; if it is "UpdateOne" then set write_func= _UpdateOneWriteFunc, and document this behavior.